tests/conftest.py

```diff
@@ -428,3 +429,44 @@ def hg_clone(hg_server, tmpdir):
     clone_dir = tmpdir.join("hg_clone")
     subprocess.run(["hg", "clone", hg_server, clone_dir.strpath], check=True)
     return clone_dir
+
+
+@pytest.fixture
+def register_codefreeze_uri(request_mocker):
+    request_mocker.register_uri(
+        "GET",
+        "https://product-details.mozilla.org/1.0/firefox_versions.json",
+        json={
+            "NEXT_SOFTFREEZE_DATE": "2122-01-01",
+            "NEXT_MERGE_DATE": "2122-01-01",
+        },
+    )
+
+
+@pytest.fixture
+def codefreeze_datetime(request_mocker):
+    today = datetime(2000, 1, 5, 0, 0, 0, tzinfo=timezone.utc)
+    freeze_date = datetime(2000, 1, 3, 0, 0, 0)
+    merge_date = datetime(2000, 1, 6, 0, 0, 0)
+    outside_freeze_date = datetime(2000, 2, 3, 0, 0, 0)
+    outside_merge_date = datetime(2000, 2, 6, 0, 0, 0)
+
+    class Mockdatetime:
+        @classmethod
+        def now(cls, tz):
+            return today
+
+        @classmethod
+        def strptime(cls, date_string, fmt):
+            if date_string == "freeze_date -0800":
+                return freeze_date
+            elif date_string == "merge_date -0800":
+                return merge_date
+            elif date_string == "outside_freeze_date -0800":
+                return outside_freeze_date
+            elif date_string == "outside_merge_date -0800":
```

Comment:
nit (to match what you have in `transplants.warning_code_freeze`)

```suggestion
            utc_offset = "-0800"
            if date_string == f"freeze_date {utc_offset}":
                return freeze_date
            elif date_string == f"merge_date {utc_offset}":
                return merge_date
            elif date_string == f"outside_freeze_date {utc_offset}":
                return outside_freeze_date
            elif date_string == f"outside_merge_date {utc_offset}":
```